pkg/backup.go

```diff
@@ -236,15 +236,26 @@ func (opt *mongoOptions) backupMongoDB() (*restic.BackupOutput, error) {
 				"--archive",
 			}, adminCreds...),
 		}
+		userArgs := strings.Fields(opt.mongoArgs)
+
 		if isStandalone {
 			backupOpt.StdinPipeCommand.Args = append(backupOpt.StdinPipeCommand.Args, "--port="+fmt.Sprint(appBinding.Spec.ClientConfig.Service.Port))
 		} else {
 			// - port is already added in mongoDSN with replicasetName/host:port format.
 			// - oplog is enabled automatically for replicasets.
-			backupOpt.StdinPipeCommand.Args = append(backupOpt.StdinPipeCommand.Args, "--oplog")
+			// Don't use --oplog if user specify any of these arguments through opt.mongoArgs
+			// 1. --db
+			// 2. --collection
+			// xref: https://docs.mongodb.com/manual/reference/program/mongodump/#cmdoption-mongodump-oplog
+			forbiddenArgs := []string{"--db", "--collection"}
+			if validArgs(userArgs, forbiddenArgs) {
+				backupOpt.StdinPipeCommand.Args = append(backupOpt.StdinPipeCommand.Args, "--oplog")
+			}
 		}
 		if opt.mongoArgs != "" {
```

Comment:
`if opt.mongoArgs != "" {` This check is not needed. `userArgs` will be empty.

Reply:
removed